pkg/api/pod/util.go

```diff
@@ -646,13 +646,20 @@ func dropDisabledFields(
 		})
 	}
 
-	// If the feature is disabled and not in use, drop the hostUsers field.
-	if !utilfeature.DefaultFeatureGate.Enabled(features.UserNamespacesSupport) && !hostUsersInUse(oldPodSpec) {
+	// If UserNamespacesSupport is disabled, or if ProcMountType is disabled and hostUsers is not in use,
+	// drop the hostUsers field and ensure ProcMount is not set to "Unmasked" for any containers.
+	if !utilfeature.DefaultFeatureGate.Enabled(features.UserNamespacesSupport) || !utilfeature.DefaultFeatureGate.Enabled(features.ProcMountType) && !hostUsersInUse(oldPodSpec) {
 		// Drop the field in podSpec only if SecurityContext is not nil.
 		// If it is nil, there is no need to set hostUsers=nil (it will be nil too).
 		if podSpec.SecurityContext != nil {
 			podSpec.SecurityContext.HostUsers = nil
 		}
+		VisitContainers(podSpec, AllContainers, func(c *api.Container, _ ContainerType) bool {
+			if c.SecurityContext != nil && c.SecurityContext.ProcMount != nil {
+				c.SecurityContext.ProcMount = nil
```

Comment:
Is setting this to nil the right thing?   based on the comment in https://github.com/kubernetes/kubernetes/blob/5fc7032a0e509d41890573d5ea2beec880593bda/pkg/api/pod/util.go#L897-L900

Is it possible for this to be a different value here?